filebeat/harvester/reader/multiline.go

```diff
@@ -260,8 +282,14 @@ func (mlr *Multiline) addLine(m Message) {
 	mlr.readBytes += m.Bytes
 }
 
-func (mlr *Multiline) reset() {
-	mlr.state = (*Multiline).readFirst
+// reset sets state of the reader to readFirst
+func (mlr *Multiline) resetState() {
+	mlr.setState((*Multiline).readFirst)
+}
+
+// setState sets state to the given function
+func (mlr *Multiline) setState(next func(mlr *Multiline) (Message, error)) {
```

Comment:
let's change state handlers from methods to functions: 

`func (m *Multiline) readX() (Message, error)` to `func readX(m *Multiline) (Message, error)`. Both are equivalent when used as function pointers, but we can write `mlr.state = readFirst` instead of `mlr.state = (*Multiline).readFirst`.